src/configure/services/source/sourceRepositoryService.ts

```diff
@@ -0,0 +1,183 @@
+                    repositoryName: AzureDevOpsService.getRepositoryNameFromRemoteUrl(remoteUrl),
+                    remoteUrl: remoteUrl,
+                    branch: branch,
+                    commitId: commitId,
+                    localPath: repositoryPath
+                };
+            }
+            else if (GitHubProvider.isGitHubUrl(remoteUrl)) {
+                let repoId = GitHubProvider.getRepositoryIdFromUrl(remoteUrl);
+                return <GitRepositoryParameters>{
+                    repositoryProvider: RepositoryProvider.Github,
+                    repositoryId: repoId,
+                    repositoryName: repoId,
+                    remoteUrl: remoteUrl,
+                    branch: branch,
+                    commitId: commitId,
+                    localPath: repositoryPath
+                };
+            }
+            else {
+                throw new Error(Messages.cannotIdentifyRespositoryDetails);
+            }
+        }
+        else {
+            throw new Error(Messages.remoteRepositoryNotConfigured);
+        }
+    }
+
+    /**
+     *
+     * @param pipelineYamlPath : local path of yaml pipeline in the extension
+     * @param context: inputs required to be filled in the yaml pipelines
+     * @returns: thenable object which resolves once all files are added to the repository
+     */
+    public addYmlFileToRepo(pipelineTemplateFilePath: string, repoPath: string, context: WizardInputs): Q.Promise<string> {
+        let deferred: Q.Deferred<string> = Q.defer();
+        fs.readFile(pipelineTemplateFilePath, { encoding: "utf8" }, async (error, data) => {
+            if (error) {
+                throw new Error(error.message);
+            }
+            else {
+                let fileContent = Mustache.render(data, context);
+                let ymlPipelineFileName: string = await SourceRepositoryService.getPipelineFileName(repoPath);
+                let ymlFileUri = vscode.Uri.file(path.join(repoPath, "/" + ymlPipelineFileName));
+                fs.writeFileSync(ymlFileUri.fsPath, fileContent);
+                await vscode.workspace.saveAll(true);
+                deferred.resolve(ymlPipelineFileName);
+            }
+        });
+
+        return deferred.promise;
+    }
+
+    /**
+     * commits yaml pipeline file into the local repo and pushes the commit to remote branch.
+     * @param pipelineYamlPath : local path of yaml pipeline in the repository
+     * @returns: thenable object which resolves once commit is pushed to remote branch, and failure message if unsuccessful
+     */
+    public async commitAndPushPipelineFile(pipelineYamlPath: string): Promise<{ commitId: string, branch: string }> {
+        // TODO
+        // unstages the changes already staged
+        // stages the file at the mentioned path
+        // commit the staged file
+        // stage the previously staged changes
+
+
+        await this.gitReference.add(pipelineYamlPath);
```

Comment:
Some spec questions around if there are more files which are edited/staged by user. What should happen